cmd/ko/README.md

```diff
@@ -190,6 +186,31 @@ to whatever `kubectl` context is active.
 of convenience for cleaning up resources created through `ko apply`.
 
 
+## With DockerHub
+
+Unfortunately, DockerHub does not support this sort of multi-level names, so you may see an error like:
+
+```shell
+$ KO_DOCKER_REPO=docker.io/mattmoor ko publish ./cmd/crane
+2018/07/19 03:25:56 Using base gcr.io/distroless/base:latest for github.com/google/go-containerregistry/cmd/crane
+2018/07/19 03:25:56 Publishing index.docker.io/mattmoor/github.com/google/go-containerregistry/cmd/crane:latest
+2018/07/19 03:25:57 error publishing github.com/google/go-containerregistry/cmd/crane: UNAUTHORIZED: "authentication required"
+```
+
+To support this, we have a flag that will flatten the import path to `{package}-{hash of import path}`:
```

Comment:
What do you think about hard-coding the default value for this based on the registry? It feels impure but I think it's not that unreasonable to do for docker hub and gcr. Using the flag should override it and it wouldn't be hard to add other registries as needed.